app/models/spree_amazon/order.rb

```diff
@@ -44,8 +44,16 @@ def close_order_reference!
     end
   end
 
-  def save_total
-    mws.set_order_data(total, currency)
+  def set_order_reference_details(total, options={})
+    # These are parameters that the Amazon gem accepts
+    options.assert_valid_keys(
```

Comment:
Hm, why don't we call it `amazon_options` and skip the assertion?  Then it's clear that the list of valid options comes from Amazon?